### PR TITLE
New: add fromThrowable method in either monad

### PR DESCRIPTION
**Pull request by w3slley**
Implementation for #7.
Also added .prettierrc to the project

### CODE REVIEW

File: package-lock.json
Comment (w3slley) [inline package-lock.json:1]: Not sure if we should add this to the project, let me know what you think

**Comment by w3slley** [comment]
We can also use deno 2.0, it has linting, formatting, typescript support among other things out of the box

File: src/functional/either.ts [hunk 1]
```diff
@@ -66,10 +66,10 @@ export interface Either<O, E> {
      * @param matcher Object containing where functions for Ok and Error.
      * @returns The Either of the where function.
      */
-    where<T,R>(matcher: {Ok: (value: O) => T, Error: (value: E) => R}): T | R
-    where<T,R>(matcher: {Ok: T, Error: (value: E) => R}): T | R
-    where<T,R>(matcher: {Ok: (value: O) => T, Error: R}): T | R
-    where<T,R>(matcher: {Ok: T, Error: R}): T | R
+    where<T, R>(matcher: { Ok: (value: O) => T; Error: (value: E) => R }): T | R
+    where<T, R>(matcher: { Ok: T; Error: (value: E) => R }): T | R
```
Comment (w3slley) [inline src/functional/either.ts:70]: prettier changed this :)

File: src/functional/either.ts [hunk 2]
```diff
@@ -226,12 +232,21 @@ class __Either {
...
     static fromNullable<T, R>(value: T, defaultVal: R): Error<R> | Ok<T> {
-        if(value == null) {
+        if (value == null) {
             return Error.of(defaultVal)
         }
 
         return Ok.of(value)
     }
+
+    static fromThrowable<T>(fn: () => T): Error<unknown> | Ok<T> {
```
Comment (w3slley) [inline src/functional/either.ts:242]: I used `unknown` but was unsure if we could use a type here

Comment (DrKillshot) [reply src/functional/either.ts:242]: I think `unknown` is the right call here.

**Review by DrKillshot — APPROVED** [review]

**Pull request merged**